src/main/java/com/exactpro/th2/sailfish/utils/filter/CompareFilter.java

```diff
@@ -156,17 +161,52 @@ private int compareValueWithoutOperation(Comparable<?> first, Comparable<?> seco
         if (second instanceof LocalDate || second instanceof LocalDateTime || second instanceof LocalTime) {
             throw new IllegalArgumentException(String.format("Failed to compare Temporal values {%s}, {%s}", first, second));
         }
-        if (first instanceof BigDecimal) {
-            return ((BigDecimal)first).compareTo(new BigDecimal(second.toString()));
+        var result = compareWithTransformation(BigDecimal.class, first, second, it -> new BigDecimal(it.toString()));
+        if (result.matchType) {
+            return result.comparisonResult;
         }
-        if (second instanceof BigDecimal) {
-            return new BigDecimal(first.toString()).compareTo((BigDecimal)second);
+        result = compareWithTransformation(BigInteger.class, first, second, it -> new BigInteger(it.toString()));
+        if (result.matchType) {
+            return result.comparisonResult;
         }
-        return ((Long)first).compareTo((Long)second);
+        return Long.valueOf(first.toString()).compareTo(Long.valueOf(second.toString()));
```

Comment:
Maybe add personal handler for Number types and trow errors at the end of this method to notify about unsupported type.
I think we can write more clear text than NumberFormat exception